R/utils_visr.R

```diff
@@ -267,3 +267,143 @@ legendopts <- function(legend_position = "right",
 
 }
 
+#' @title Extract the numerical alpha representation of #RRGGBBAA colour 
```

Comment:
why do we need below functions? They seem very similar. Could we not make one big function? Perhaps I misunderstood?

Reply:
ggplot seems to use a mixture of a column called "alpha" and appending the alpha value to a #RRGGBB colour in the "colour" column to adjust the alpha values, depending on the user behaviour. Therefore I needed helper functions to convert between numerical and hex-encoded alpha values both ways (`convert_alpha()`), extract the numerical alpha value of a #RRGGBBAA colour to apply my modifier to it (`get_alpha_from_hex_colour()`) and a convencience wrapper to replace the AA part of a #RRGGBBAA colour (`replace_hex_alpha()`). Some of the functions use each other, but cramming all the functionality into one big function would make it very cumbersome to use.

Reply:
I would explain this behaviour in a bit more detail in the description so we remember this and users understand it as well.